chingu-voyages/v41-bears-team-20
# Connect frontend with backend

Literrally just copy pasted code from the [implement-auth](https://github.com/chingu-voyages/v41-bears-team-20/tree/implement-auth)

It doesn't work atm. I think it will take me 2 more days to get it done

---

Edit:

**HOW TO TEST**
- Setup:
  - clone pull request `git fetch origin refs/pull/83/head:pr-83`
  - switch to branch: `git checkout pr-83`
  - Run `npm i`
  - Run `cd server`
  - Run `npm run db:docker:up`
  - Run `docker ps` to confirm containers are running
  - Run `npx prisma migrate` or `npx prisma reset`
  - Run `npm run start:dev:`
  - Open new terminal instance
  - Run `cd server`
  - Run `npx prisma studio` --> opens up database viewer
  - Open new terminal instance
  - Run `cd client`
  - Run `npm run dev`
  - Navigate to `http://localhost:5173/account/register` in the browser

- Sign up for account
  - See if password policy violation displays an error message
  - Inspect console.log, network tab, and application> localstorage for confirmation of sign up
  - See if you were navigated to home page that displays "Hello <user id> " message
  - See if the navbar changed to display sign out button

- Sign out
  - See if navbar changes and you are navigated to home page that displays "Home page"
  
- Sign in
  - go to `http://localhost:5173/account/login`
  - login with recently created account credentials
  - Follow same steps as `sign up for account` 
  
**Demos**

_Signup_
![image](https://user-images.githubusercontent.com/14286113/200851270-762325a0-e48f-40b1-90de-3c8c7abfcfb1.png)
![image](https://user-images.githubusercontent.com/14286113/200851383-a06169c2-9931-4898-aba2-ea85f7044a97.png)
